airsonic-main/src/main/java/org/airsonic/player/controller/StreamController.java

```diff
@@ -230,25 +232,37 @@ public void handleRequest(HttpServletRequest request, HttpServletResponse respon
             ) {
                 final int BUFFER_SIZE = 2048;
                 byte[] buf = new byte[BUFFER_SIZE];
+                long bytesWritten = 0;
 
                 while (!status.terminated()) {
                     if (player.getPlayQueue().getStatus() == PlayQueue.Status.STOPPED) {
                         if (isPodcast || isSingleFile) {
                             break;
                         } else {
-                            sendDummy(buf, out);
+                            sendDummyDelayed(buf, out);
                         }
                     } else {
 
                         int n = in.read(buf);
                         if (n == -1) {
                             if (isPodcast || isSingleFile) {
+                                // Pad the output if needed to avoid content length errors on transcodes
+                                if (fileLengthExpected != null && bytesWritten < fileLengthExpected) {
+                                    sendDummy(buf, out, fileLengthExpected - bytesWritten);
+                                }
                                 break;
                             } else {
-                                sendDummy(buf, out);
+                                sendDummyDelayed(buf, out);
```

Comment:
out of curiosity, why is this sent after a delay?

Reply:
I just preserved the original behavior, renaming the function so its name was a better indication of what it does.  I surmise it is attempting to keep the stream alive while waiting for some other event to occur, but I'm not really sure under what circumstances it was used.

Reply:
okay, i'll try to dig in what's happening